[ethantyh692] iP

ethantyh692 opened the pull request:
### Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ethantyh692 committed:
Done with Level-0

ethantyh692 committed:
Done with Level-1

ethantyh692 committed:
Done with Level-2

ethantyh692 committed:
Done with Level-3, added Task class

ethantyh692 committed:
Done with Level-4, added Todo, Event and Deadline classes

ethantyh692 committed:
Done with A-TextUiTesting

ethantyh692 committed:
Done with Level-5, added exceptions

ethantyh692 committed:
Done with Level-6, added InvalidIndexException

ethantyh692 committed:
Add save functionality

ethantyh692 committed:
Merge branch 'branch-Level-7'

ethantyh692 committed:
Add ability to handle datetime instances and cleaned up exception handling

ethantyh692 committed:
Merge remote-tracking branch 'origin'

ethantyh692 committed:
Modularise into different classes

ethantyh692 committed:
Merge branch 'branch-A-MoreOOP'

ethantyh692 committed:
Add duke package with all java files

ethantyh692 committed:
Merge remote-tracking branch 'origin'

ethantyh692 committed:
Merge branch 'add-gradle-support'

ethantyh692 committed:
Add JUnit test cases for Parser and TaskList

ethantyh692 committed:
Merge branch 'branch-A-JUnit'

ethantyh692 committed:
Create JAR File

ethantyh692 committed:
Add JavaDoc comments to the code

ethantyh692 committed:
Configure to coding standards

ethantyh692 committed:
Add Find function

ethantyh692 committed:
Merge branch 'branch-A-JavaDoc'

ethantyh692 committed:
Merge branch 'branch-A-CodingStandard'

ethantyh692 committed:
Add Find function

ethantyh692 committed:
Merge

ethantyh692 committed:
Add a GUI to Duke

ethantyh692 committed:
Merge branch 'branch-Level-10'

chrisjohntan commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,57 @@
+
+public class TaskListTest {
+
+    private final Path dataDirectory = Paths.get("data");
+    private final File testFile = new File(dataDirectory.toFile(), "test.txt");
+    private final PrintStream originalOut = System.out;
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
```
Would it be better for the names of these constants to be uppercased?

chrisjohntan commented on src/test/java/duke/TaskListTest.java:
```diff
@@ -0,0 +1,57 @@
+package duke;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
```
Import statements are well organised and consistent 👍

chrisjohntan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,228 @@
+        return null;
+    }
+
+    private Todo parseTodoCommand(String command) {
+        String description = command.substring(4).trim(); // Extract description
+        if (description.isEmpty()) {
+            throw new IllegalArgumentException("Description for 'todo' cannot be empty.");
+        }
+        return new Todo(description);
+    }
+
+    private Deadline parseDeadlineCommand(String command) {
+        String[] parts = command.split("/by");
+        if (parts.length != 2) {
+            throw new IllegalArgumentException(
+                    "Deadline command must contain '/by' followed by a date-time.");
+        }
+
+        String description = parts[0].substring(8).trim();
+        if (description.isEmpty()) {
+            throw new IllegalArgumentException("Description for 'deadline' cannot be empty.");
+        }
+
+        String dateTimeString = parts[1].trim();
+        LocalDateTime by = convertStringToDate(dateTimeString);
+        return new Deadline(description, by);
+    }
+
+    protected Event parseEventCommand(String command) {
+        String[] partsFrom = command.split("/from");
+        if (partsFrom.length != 2) {
```
I like the consistent naming of you methods! 💯

chrisjohntan reviewed (commented):
LGTM 🚀

ylchin reviewed (commented):
good naming of variables and javadocs. import statements are in compliance with standards too.

ethantyh692 committed:
Add assertions

There are currently no assertions.

This needs to change because there is a need to verify inputs.

Add assertions to verify inputs of key methods.

It is done as such so that we ensure no null inputs are passed.

ethantyh692 committed:
Improve Code Quality

Some parts of the current code do not follow SLAP.

It compromises readability and makes debugging harder.

Add some layers of abstraction.

This is done to fulfil SLAP.

ethantyh692 committed:
Merge pull request #1 from ethantyh692/branch-A-Assertions

Add assertions

ethantyh692 committed:
Resolve merge confict between master and branch-A-CodeQuality

ethantyh692 committed:
Resolve a few more changes

ethantyh692 committed:
Merge branch 'master' into branch-A-CodeQuality

ethantyh692 committed:
Merge pull request #2 from ethantyh692/branch-A-CodeQuality

Improve Code Quality

ethantyh692 committed:
Add priority function

ethantyh692 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/duke/Parser.java

ethantyh692 committed:
Resolve merge conflicts between master and branch-BCD-Extension

ethantyh692 committed:
Add user guide

ethantyh692 committed:
Add final changes

ethantyh692 committed:
Add last changes
